misc/swagger-ballerina/modules/swagger-to-ballerina-generator/src/main/java/org/ballerinalang/swagger/model/BalOpenApi.java

```diff
@@ -35,41 +39,100 @@
  * Wrapper for <code>io.swagger.oas.models.OpenAPI</code>
  * <p>This class can be used to push additional context variables for handlebars</p>
  */
-public class OpenApiWrapper {
-
+public class BalOpenApi {
     private String apiPackage;
     private String openapi = "3.0.0";
     private Info info = null;
     private ExternalDocumentation externalDocs = null;
     private List<Server> servers = null;
-    private List<SecurityRequirement> security = null;
+    private Set<Map.Entry<String, String>> security = null;
     private List<Tag> tags = null;
     private Set<Map.Entry<String, PathItem>> paths = null;
+    private Set<Map.Entry<String, BalSchema>> schemas = null;
     private Components components = null;
     private Map<String, Object> extensions = null;
     private String host = null;
     private int port = 0;
     private int httpsPort = 0;
     private String basePath = null;
-
-    public OpenApiWrapper buildFromOpenAPI(OpenAPI openAPI) throws MalformedURLException {
+    private String url = null;
+
+    /**
+     * Build a {@link BalOpenApi} object from a {@link OpenAPI} object.
+     * All non iterable objects using handlebars library is converted into
+     * supported iterable object types.
+     *
+     * @param openAPI {@link OpenAPI} type object to be converted
+     * @return Converted {@link BalOpenApi} object
+     * @throws BalOpenApiException when OpenAPI to BalOpenApi parsing failed
+     */
+    public BalOpenApi buildFromOpenAPI(OpenAPI openAPI) throws BalOpenApiException {
         this.openapi = openAPI.getOpenapi();
         this.info = openAPI.getInfo();
         this.externalDocs = openAPI.getExternalDocs();
         this.servers = openAPI.getServers();
-        this.security = openAPI.getSecurity();
         this.tags = openAPI.getTags();
         this.components = openAPI.getComponents();
         this.extensions = openAPI.getExtensions();
         this.paths = openAPI.getPaths().entrySet();
 
+        try {
+            setSecurityRequirements(openAPI);
+            setHostInfo(openAPI);
+            setSchemas(openAPI);
+        } catch (MalformedURLException e) {
+            throw new BalOpenApiException("Failed to parse server information", e);
+        }
+
+        return this;
+    }
+
+    /**
+     * Populate schemas into a "Set".
+     *
+     * @param openAPI <code>OpenAPI</code> definition object with schema definition
+     */
+    private void setSchemas(OpenAPI openAPI) {
+        this.schemas = new LinkedHashSet<>();
+        Map<String, Schema> schemaMap;
+        if (openAPI.getComponents() == null) {
+            return;
+        }
+        if (openAPI.getComponents().getSchemas() == null) {
```

Comment:
Can we merge these two if conditions together?

Reply:
Fixed with 06d87e1